### PR TITLE
New mimic play page

### PR DESCRIPTION
**Pull request by Houra-Am**
### Motivation

Mimic statistics improvement. Rendering stats based on choice.

### Changes
MimicStats.tsx
 Choice selection useMemo added.

### Test

When playing, chose different response to check the stats.

### CODE REVIEW

File: src/components/activities/MimicStats.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        <>
+          {responseChoices.map((responseChoice) => {
+            if (responseChoice === 0) {
+              return <LinearProgressWithLabel value={stats[0]} style={{ height: '0.5rem', margin: '1.2rem 0' }} />;
+            } else if (responseChoice === 1) {
+              return <LinearProgressWithLabel value={stats[1]} style={{ height: '0.5rem', margin: '1.2rem 0' }} />;
+            } else {
+              return <LinearProgressWithLabel value={stats[2]} style={{ height: '0.5rem', margin: '1.2rem 0' }} />;
+            }
+          })}
```
Comment (DavidRobertAnsart) [inline src/components/activities/MimicStats.tsx:65]: **nit:** You can replace this if/else by using directly responseChoice as the array index!

**Suggestion:**
```tsx
{responseChoices.map((responseChoice) => (
    <LinearProgressWithLabel value={stats[responseChoice]} style={{ height: '0.5rem', margin: '1.2rem 0' }} />
))}
```

**Review by DavidRobertAnsart — APPROVED** [review]
Perfect! 🚀 🔥

File: src/components/activities/MimicStats.tsx (hunk 1 above)
Comment (Houra-Am) [reply src/components/activities/MimicStats.tsx]: It is indeed cleaner. I had to add key prop though.

Comment (Houra-Am) [reply src/components/activities/MimicStats.tsx]: It is indeed a cleaner way. I had to add  a key prop though.

**Review by DavidRobertAnsart — APPROVED** [review]

**Pull request merged**